metagraph/src/graph/representation/masked_graph.hpp

```diff
@@ -119,6 +119,7 @@ class MaskedDeBruijnGraph : public DeBruijnGraph {
     virtual void set_mask(bitmap *mask) { kmers_in_graph_.reset(mask); }
 
     virtual const bitmap& get_mask() const { return *kmers_in_graph_; }
+    virtual std::unique_ptr<bitmap>& get_mask_ptr() { return kmers_in_graph_; }
```

Comment:
Wouldn't it be better to construct a new masked graph, instead of doing these hacks with modifying an existing mask? (you can re-use the same graph, so the overhead will be zero).

If you modify the mask, it may break the internal properties of the graph (currently only `only_valid_nodes_in_mask_`).

Reply:
But there is overhead since you have to create a new bit vector to populate

Reply:
This is an abstract bitmap, which is not necessarily modifiable, anyway.
Copying a bitmap is usually acceptable.